Heat_method_3/doc/Heat_method_3/Heat_method_3.txt

```diff
@@ -0,0 +1,249 @@
+model of the concept `SparseLinearAlgebraWithFactorTraits_d`.
+This implementation is based on \cgalCite{cgal:cww-ghnac-13} and \cgalCite{cgal:fsbs-acidt-06}
+
+
+
+
+\section sec_HM_examples Examples
+
+We give examples for the free function `CGAL::Heat_method_3::estimate_geodesic_distances()`,
+for the class template `CGAL::Heat_method_3::Surface_mesh_geodesic_distances_3`, with and without the use
+of intrinsic Delaunay triangulation.
+
+\subsection HM_example_Free_function Using a Free Function
+
+The first example calls the free function `Heat_method_3::estimate_geodesic_distances()`,
+which computes for all vertices of a triangle mesh the distances to a single source vertex.
+
+The distances are written into an internal property map of the surface mesh.
+
+\cgalExample{Heat_method_3/heat_method.cpp}
+
+For a `Polyhedron_3` you can either add a data field to the vertex type, or, as shown
+in the following example, create a `boost::unordered_map` and pass it to the function
+`boost::make_assoc_property_map()`, which generates a vertex distance property map.
+
+\cgalExample{Heat_method_3/heat_method_polyhedron.cpp}
+
+
+\subsection HM_example_Class Using the Heat Method Class
+
+The following example shows the heat method class. It can be used
+when one adds and removes source vertices. It precomputes matrix
+factorization, which depend only on the input mesh and not the particular
+set of source vertices.  In the example we compute the distances to one
+source, add the farthest vertex as a second source vertex, and then compute
+the distances with respect to these two sources.
+
+\cgalExample{Heat_method_3/heat_method_surface_mesh.cpp}
+
+
+\subsection HM_example_Intrinsic Switching off the Intrinsic Delaunay Triangulation
+
+The following example shows the heat method on a triangle mesh without using the
+intrinsic Delaunay triangulation (iDT) algorithm, for example because by constrution
+your meshes have a good quality (Poor quality in this case means that the input
+is far from Delaunay, though even in this case one may still get good results without iDT,
+depending on the specific geometry of the surface.)
+
+\cgalExample{Heat_method_3/heat_method_surface_mesh_direct.cpp}
+
+
+
+\section sec_HM_definitions Theoretical Background
+
+Section \ref Subsection_HM_Definitions_Intro gives an overview of the theory needed by the Heat method.
+Section \ref Subsection_HM_IDT_Definitions gives the background needed for the Intrinsic Delaunay triangulation.
+
+\subsection Subsection_HM_Definitions_Intro The Heat Method Algorithm
+
+For a detailed overview of the heat method, the reader may consult
+\cgalCite{cgal:cww-ghnac-13} to read the original article. In the
+sequel, we introduce the basic notions so as to explain our
+algorithms. In general, the heat method is applicable to any setting
+if there exists a gradient operator \f$ \nabla\f$, a divergence
+operator \f$\nabla\f$ and a Laplace operator \f$\Delta = \nabla \cdot
```

Comment:
Gradient and divergence operator have the same notation, is it expected? In the following L seems to be used for Laplacien.

Reply:
Ping

Reply:
I am sorry, but even in the [TOG paper](https://www.cs.cmu.edu/~kmcrane/Projects/HeatMethod/paperCACM.pdf) you have twice the same symbol.  I have to wait for Keenan's reply to a mail.

Reply:
- The gradient of a function `f` is usually noted `∇f`.
- The divergence operator is sometimes noted `∇·` is the scalar product. That is a strange notation, because in a way, the divergence is the scalar product of the gradient with something.
- That way, the Laplacian of f (divergence of the gradient) is `∇·∇f`.

Reply:
@afabri I suggest you synchronize with @lrineau and do the fix.